services/datamanager/builtin/builtin.go

```diff
@@ -517,14 +542,23 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 	// Initialize or add a collector based on changes to the component configurations.
 	newCollectorMetadata := make(map[componentMethodMetadata]bool)
 	for _, attributes := range allComponentAttributes {
-		if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 {
+		if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 && !svc.captureDisabled {
 			componentMetadata, err := svc.initializeOrUpdateCollector(
 				attributes, updateCaptureDir)
 			if err != nil {
 				svc.logger.Errorw("failed to initialize or update collector", "error", err)
 			} else {
 				newCollectorMetadata[*componentMetadata] = true
 			}
+		} else if attributes.Disabled {
+			// if disabled, make sure that it is closed, so it doesn't keep collecting data.
+			collector, md, err := svc.getCollectorFromConfig(attributes)
+			if err != nil {
+				svc.logger.Errorw("collector ", attributes.Name, " was not found", "error", err)
```

Comment:
Is this necessarily an error? Or would it occur with any component that hasn't and doesn't have data capture turned on? If the latter (or some other valid case), it should probably be an INFO level log

Reply:
that makes sense! changing to info